plugins/kubernetes-backend/src/service/KubernetesProxy.ts

```diff
@@ -138,7 +139,7 @@ export class KubernetesProxy {
         ws: true,
         secure: !originalCluster.skipTLSVerify,
         changeOrigin: true,
-        pathRewrite: { [`^${originalReq.baseUrl}`]: '' },
+        pathRewrite: { [`^${originalReq.baseUrl}`]: originUrl.pathname || '' },
```

Comment:
... so if the cluster's URL has changed since the last time a request was proxied to it, this path rewrite will use the old URL. Not sure what impact this has, but it seems like a more robust way to achieve what you want here would be to use a [path rewrite function](https://github.com/chimurai/http-proxy-middleware/blob/master/recipes/pathRewrite.md#custom-rewrite-function). Not sure if it plays nicely with async functions though, and that could be a problem since the rewrite function would need to call `getClusterForRequest` on the current request to get the up-to-date cluster details.

Reply:
did a quick investigation and it looks like an async path rewrite function should work! https://github.com/chimurai/http-proxy-middleware/blob/d7aa01de280d598537735733070ad06d9ea608dd/src/http-proxy-middleware.ts#L161

Reply:
This is helpful